### PR TITLE
[#3] Setup CD

### LINKED ISSUE #3: Setup CD
## Why

To align with Nimble's workflow and ensure the effectiveness of the product delivery we should have automated CD

## Acceptance Criteria

- [ ] Setup CD on Github Action to deploy on [Netlify](https://www.netlify.com/)

### PR DESCRIPTION
**Pull request by carryall**
Resolve https://github.com/carryall/react-survey-challenge/issues/3

## What happened 👀

- Set auth token and site id in repository secrets
  Screen Shot 2564-10-12 at 15 29 27
- Add deployment workflows using [Netlify Action](https://github.com/marketplace/actions/netlify-actions)
  - Preview deployment on pull request
    Screen Shot 2564-10-12 at 17 02 01
  - Branch deployment on `develop` branch
    Screen Shot 2564-10-12 at 17 01 43
  - Production deployment on `master` branch
    Screen Shot 2564-10-12 at 17 01 51

Note:
- Branch deployment and production deployment would trigger only when the test workflows completed
- The `workflow_run` condition on the workflow is not testable until it's on the main branch of the repository

## Insight 📝

N/A

## Proof Of Work 📹

This [PR deployment](https://deploy-preview-28--nimble-survey.netlify.app/) works

### CODE REVIEW

File: .github/workflows/deploy-preview.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
```
Comment (carryall) [inline .github/workflows/deploy-preview.yml:18]: Cancel previous run from the same branch if any

File: .github/workflows/deploy-preview.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  pull_request:
+
+jobs:
+  deploy-preview:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
```
Comment (hoangmirs) [inline .github/workflows/deploy-preview.yml:12]: The same comment as CI PR, do we need it here? And should we change to use `setup-node` action?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+name: Deploy
+
+on:
+  workflow_run:
+    workflows:
+      - Unit Test
```
Comment (hoangmirs) [inline .github/workflows/deploy.yml]: I think maybe we need to rename the test workflows to start with a verb like the workflows on this PR. And compass also suggests using it:
https://nimblehq.co/compass/development/code-conventions/github-actions/#naming

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      - UI Test
+    branches:
+      - master
+      - development
```
Comment (hoangmirs) [inline .github/workflows/deploy.yml]: I think it should be `main` and `develop` as you set them at the moment.

```suggestion
    branches:
      - main
      - develop
```

Or they can be detected automatically with those keywords 🤔

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (carryall) [reply .github/workflows/deploy.yml]: Renamed in 75dcd62

File: .github/workflows/deploy-preview.yml (hunk 2 above)
Comment (carryall) [reply .github/workflows/deploy-preview.yml:12]: Added node setup step on 53d6bbc

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (carryall) [reply .github/workflows/deploy.yml]: Fixed in 7e224f0

**Review by hoangmirs — APPROVED** [review]

**Review by rafayet-monon — APPROVED** [review]

**Pull request merged**